src/RageSoundMixBuffer.cpp

```diff
@@ -4,17 +4,29 @@
 
 #include <cmath>
 #include <cstdint>
+#include <cstdlib>
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_iBufSize = m_iBufUsed = 0;
-	m_pMixbuf = nullptr;
+	// See how many samples we can stuff into 2MB.
+	size_t BUF_SIZE = 2 * 1024 * 1024 / sizeof(float);
+	
+	m_pMixbuf = static_cast<float*>(std::malloc(BUF_SIZE));
+	if (m_pMixbuf == nullptr) {
+		ASSERT_M(false, "Failed to allocate memory for the sound mixing buffer");
+	}
+
+	// To ensure the sound buffer doesn't contain any garbage, this will
+	// initialize the sound buffer to be completely empty.
+	std::memset(m_pMixbuf, 0, m_iBufSize * sizeof(float));
```

Comment:
Hmm I still don't see `m_iBufSize` being initialized/non-zero. It's not set in this constructor.

Reply:
Oops totally misread the comment last time you reviewed this